### PR TITLE
Fix WaitForPersistentVolumeClaimPhase returning error unconditionally

### PR DESCRIPTION
**Pull request by copejon**
Fix function returning an error regardless of the output from the `wait` call.

### CODE REVIEW

File: tests/utils/pvc.go [hunk 1]
```diff
@@ -76,12 +76,15 @@ func NewPVCDefinition(pvcName string, size string, annotations, labels map[strin
...
+	err := wait.PollImmediate(pvcPollInterval, PVCPhaseTime, func() (bool, error) {
 		pvc, err := clientSet.CoreV1().PersistentVolumeClaims(namespace).Get(pvcName, metav1.GetOptions{})
 		if err != nil || pvc.Status.Phase != phase {
 			return false, err
 		}
 		return true, nil
 	})
-	return fmt.Errorf("PersistentVolumeClaim %s not in phase %s within %v", pvcName, phase, PVCPhaseTime)
+	if err != nil{
+		return fmt.Errorf("PersistentVolumeClaim %s not in phase %s within %v", pvcName, phase, PVCPhaseTime)
```
Comment (jeffvance) [inline tests/utils/pvc.go]: what about:
```
func WaitForPersistentVolumeClaimPhase(clientSet *kubernetes.Clientset, namespace string, phase k8sv1.PersistentVolumeClaimPhase, pvcName string) error {
	return wait.PollImmediate(pvcPollInterval, PVCPhaseTime, func() (bool, error) {
		pvc, err := clientSet.CoreV1().PersistentVolumeClaims(namespace).Get(pvcName, metav1.GetOptions{})
		if err != nil || pvc.Status.Phase != phase {
			return false, err
		}
		return true, nil
	})
}
```

Comment (jeffvance) [reply tests/utils/pvc.go]: Although, a downside to my suggestion is that the caller doesn't see the nice err msg stating the pvc was not in the expected phase after X mins...

Comment (copejon) [reply tests/utils/pvc.go]: I'd considered that as well, but opted for the verbose error message.

Comment (jeffvance) [reply tests/utils/pvc.go]: ack. Ok as is!

**Comment by jeffvance** [comment]
closed by #378

**Pull request closed without merging**